internal/gitlab/gitlab.go

```diff
@@ -463,6 +463,18 @@ func MRList(projID interface{}, opts gitlab.ListProjectMergeRequestsOptions, n i
 		if err != nil {
 			return nil, err
 		}
+		// SourceBranch was specified, so we need to make sure we're listing MRs
+		// for the source branch for the specific projID and not a fork.
```

Comment:
That assumes that the lab user only creates merge requests from branches on the upstream remote, and not their own fork. If you are pushing to a fork and create a merge request from that, then the whole feature is now broken.

IMHO the correct ID for filtering is the project that corresponds to the branch's upstream remote.

Maybe including `AuthorID` in the options would be an alternative to filtering by branch, although that would mean that guessing doesn't work for other people's merge requests (for example for branches created via `lab mr checkout ...`).

Reply:
> Do you mean the following change?

No. As I understand it, `SourceProjectID` refers to the project where the branch that is being merged comes from, and `ProjectID` is the project where it is being merged (no clue what's the difference between `ProjectID` and `TargetProjectID`).

If that's correct, then it depends on whether the two should or should not match:

.1 Push directly to the upstream remote, IDs should match:
```sh
$ git push -u origin current-branch; lab mr create
```
2. Push to the personal fork, IDs should not match:
```sh
$ git push -u fmuellner current-branch; lab mr create
```

I think what we *really* want is based on the branch's upstream remote (like `git config branch.current-branch.remote`) and somehow match that to a gitlab project. I'm afraid I don't have any great ideas how to actually do that :-(

In particular "origin" is tricky, as I've seen two different uses:
 - 'origin' refers to the upstream
 - 'origin' refers to the personal fork, the upstream remote uses 'upstream'

We probably have to look at the remote's url. It looks like the Project type has `SSHURLToRepo` and `HTTPURLToRepo` fields, but relying on exact matches sounds fragile (I'm thinking of stuff like http vs. https, trailing slashes, ...)

Reply:
> I think what we _really_ want is based on the branch's upstream remote (like `git config branch.current-branch.remote`) and somehow match that to a gitlab project. I'm afraid I don't have any great ideas how to actually do that :-(

It turns out that all the bits that are needed are already in place, so this was a lot easier than I thought:
https://github.com/zaquestion/lab/pull/838

Reply:
Ah right! I see what you meant and how it breaks `mr create`.
Thanks for the quick PR. I'll review #838.